examples/matmul/matmul.cc

```diff
@@ -87,9 +87,10 @@ int main() {
 	multiply(queue, mat_a_buf, mat_b_buf, mat_c_buf);
 	multiply(queue, mat_b_buf, mat_c_buf, mat_a_buf);
 
-	bool verification_passed = true;
-	verify(queue, mat_a_buf, verification_passed);
-	queue.slow_full_sync(); // Wait for verification_passed to become available
+	celerity::buffer<bool> passed_buf(1);
+	verify(queue, mat_c_buf, passed_buf);
 
-	return verification_passed ? EXIT_SUCCESS : EXIT_FAILURE;
+	// The value of `passed` can differ between hosts if only part of the verification failed.
+	const auto passed = celerity::experimental::fence(queue, passed_buf).get();
+	return passed ? EXIT_SUCCESS : EXIT_FAILURE;
```

Comment:
Bug: I just noticed that this tests `buffer_snapshot::operator bool` instead of the snapshot contents! Looks like a footgun, maybe get rid of those pointer semantics?

Reply:
Removed. There were quite a few more surface API bugs in the `buffer_snapshot` definition that I fixed as well.